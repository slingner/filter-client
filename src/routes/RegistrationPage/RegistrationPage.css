.RegistrationPage {
  display: flex;
  background-image: url(./coffee3.jpg);
  box-shadow: inset 0 0 0 1000px rgba(72, 47, 34, 0.5);
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  height: 100vh;
  text-align: center;
  margin: auto;
  justify-content: center;
}
