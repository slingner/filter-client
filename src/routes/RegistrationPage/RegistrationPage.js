import React, { Component } from 'react'
import { Section } from '../../components/Utils/Utils'
import RegistrationForm from '../../components/RegistrationForm/RegistrationForm'
import './RegistrationPage.css'

export default class RegistrationPage extends Component {
  static defaultProps = {
    history: {
      push: () => {},
    },
  }

  handleRegistrationSuccess = user => {
    const { history } = this.props
    history.push('/login')
  }

  render() {
    return (
      <Section className='RegistrationPage'>
        <h2><span aria-label='coffee-emoji' role='img'>☕</span> Register <span aria-label='coffee-emoji' role='img'>☕</span></h2>
        <RegistrationForm
          onRegistrationSuccess={this.handleRegistrationSuccess}
        />
      </Section>
    )
  }
}
