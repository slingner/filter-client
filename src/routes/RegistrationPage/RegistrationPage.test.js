import React from 'react'
import ReactDOM from 'react-dom'
import RegistrationPage from './RegistrationPage'
import renderer from 'react-test-renderer';


describe(`<RegistrationPage />`, () => {
  //Smoke Testing
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <RegistrationPage 
        
      />,
      div
    );
    ReactDOM.unmountComponentAtNode(div)
  })
  //Snapshot Testing
  it('renders UI as expected', () => {
    const tree = renderer
      .create(<RegistrationPage />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})