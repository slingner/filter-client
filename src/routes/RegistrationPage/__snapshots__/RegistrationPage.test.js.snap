// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegistrationPage /> renders UI as expected 1`] = `
<section
  className="Section RegistrationPage"
>
  <form
    className="RegistrationForm"
    onSubmit={[Function]}
  >
    <div
      role="alert"
    />
    <svg
      className="filterHandBean"
    >
      filterlogohand.svg
    </svg>
    <h2
      className="welcome"
    >
       Welcome To Filter 
    </h2>
    <div
      className="full_name"
    >
      <label
        htmlFor="RegistrationForm__full_name"
      />
      <input
        className="Input "
        id="RegistrationForm__full_name"
        name="full_name"
        placeholder="Name"
        required={true}
        type="text"
      />
    </div>
    <div
      className="user_name"
    >
      <label
        htmlFor="RegistrationForm__user_name"
      />
      <input
        className="Input "
        id="RegistrationForm__user_name"
        name="user_name"
        placeholder="Username"
        required={true}
        type="text"
      />
    </div>
    <div
      className="password"
    >
      <label
        htmlFor="RegistrationForm__password"
      />
      <input
        className="Input "
        id="RegistrationForm__password"
        name="password"
        placeholder="Password"
        required={true}
        type="password"
      />
    </div>
    <button
      className="Button "
      type="submit"
    >
      Signup
    </button>
  </form>
</section>
`;
